### PR TITLE
Allow multiple exempt labels

### LINKED ISSUE #17: Support multiple exempt labels
On the stale bot, we were able to ignore a list of labels from the runs
```yml
# Only if issues that are not triaged
exemptLabels:
  - "1. to develop"
  - "2. developing"
  - "3. to review"
  - "4. to release"
  - security
```

Could be nice if we could as well :)

On another note, the description for the `exempt-issue-label` config is not really appropriate I'd say: `The label to apply when an issue is exempt from being marked stale`. I would go for something like `The label to ignore. Any issues with that label will not be processed.`

### PR DESCRIPTION
**Pull request by ryanwilsonperkin**
Fixes #17 
cc @skjnldsv as requester: Does this match what you were thinking?

**Breaking change**: This would switch the parameters from:
- `exempt-pr-label` -> `exempt-pr-labels`
- `exempt-issue-label` -> `exempt-issue-labels`

By switching to a list syntax, we can still support a single "exempt" label (eg. at my company we just have a label called "pinned" that we would use to avoid being marked stale) but can now also support many different "exempt" labels.

**Update (13/04/2020)**:
Because we [can't use arrays as arguments to the action](https://github.com/actions/stale/pull/19#issuecomment-547472717), this PR has been switched to use a comma-delimited string to indicate multiple labels.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -18,13 +18,15 @@ inputs:
   stale-issue-label:
     description: 'The label to apply when an issue is stale'
     default: 'Stale'
-  exempt-issue-label:
-    description: 'The label to apply when an issue is exempt from being marked stale'
+  exempt-issue-labels:
+    description: 'The labels to use to exempt an issue from being marked stale'
+    default: []
```
Comment (ryanwilsonperkin) [inline action.yml]: Is this valid for the action.yml file? Seems to be valid Yaml, but not sure if it does what I expect it to (result in an empty array if no argument is provided in the `with` block).

File: src/main.ts [hunk 2]
```diff
@@ -61,9 +61,9 @@ async function processIssues(
     }
 
     let staleLabel = isPr ? args.stalePrLabel : args.staleIssueLabel;
-    let exemptLabel = isPr ? args.exemptPrLabel : args.exemptIssueLabel;
+    let exemptLabels = isPr ? args.exemptPrLabels : args.exemptIssueLabels;
 
-    if (exemptLabel && isLabeled(issue, exemptLabel)) {
+    if (exemptLabels.some(exemptLabel => isLabeled(issue, exemptLabel))) {
```
Comment (ryanwilsonperkin) [inline src/main.ts]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

File: action.yml (hunk 1 above)
Comment (skjnldsv) [reply action.yml]: Sure, looks valid: https://nodeca.github.io/js-yaml/#yaml=ICBleGVtcHQtaXNzdWUtbGFiZWxzOgogICAgZGVzY3JpcHRpb246ICdUaGUgbGFiZWxzIHRvIHVzZSB0byBleGVtcHQgYW4gaXNzdWUgZnJvbSBiZWluZyBtYXJrZWQgc3RhbGUnCiAgICBkZWZhdWx0OiBbXQ==

**Comment by skjnldsv** [comment]
Awesome! Yes this is what I have in mind :)
```yml
exempt-issue-labels:
  - with team
  - with boss
  - lunch
  - with client
```
```js
{
	'exempt-issue-labels': [
		'with team',
		'with boss',
		'lunch',
		'with client'
	]
}
```

**Comment by ryanwilsonperkin** [comment]
Hey @damccorm thanks again for your help o #11
This PR proposes a breaking change to the interface added in that one, does your team have any opinion on breaking changes in this project?

**Comment by Gustry** [comment]
Oh yes please, we need this option!
Thanks for the patch.

**Comment by LekoArts** [comment]
You won't be able to write it like this as an end-user:

```yaml
exempt-issue-labels:
  - with team
  - with boss
  - lunch
  - with client
```

Because GitHub doesn't accept that as a valid format.

You can use YAML and the `|` pipe though like:

```yaml
exempt-labels: |
  not stale
  important
```

Then you need to parse the incoming labels:

```js
exemptLabels: parseLabels(core.getInput("exempt-labels"))
```

To get an array of strings:

```js
// Since GitHub doesn't allow arrays in workflow files
// You need to use YAML and | pipe
// Therefore this function splits on new lines and creates an array
function parseLabels(files: string): string[] {
  return files.split(/\r?\n/).reduce<string[]>(
    (acc, line) =>
      acc
        .concat(line.split(","))
        .filter(pat => pat)
        .map(pat => pat.trim()),
    []
  )
}
```

**Comment by qJake** [comment]
Looking forward to this in the next release!

I don't think there will be an issue since the actions are versioned, e.g. I reference this action like this:

```yaml
uses: actions/stale@v1
```

So as long as this is included in a "v2" for example, it shouldn't break anything existing.

**Comment by goanpeca** [comment]
Is this being merged soon 🙃?

File: src/main.ts [hunk 3]
```diff
@@ -61,9 +61,9 @@ async function processIssues(
     }
 
     let staleLabel = isPr ? args.stalePrLabel : args.staleIssueLabel;
-    let exemptLabel = isPr ? args.exemptPrLabel : args.exemptIssueLabel;
+    let exemptLabels = isPr ? args.exemptPrLabels : args.exemptIssueLabels;
```
Comment (goanpeca) [inline src/main.ts]:
```suggestion
    let exemptLabelsString = isPr ? args.exemptPrLabels : args.exemptIssueLabels;
    let exemptLabels = exemptLabelsString.split(',');
```

**Comment by dmodoomsirius** [comment]
I do hope this becomes a thing soon!!!

**Comment by qJake** [comment]
Eagerly awaiting this in actions/stalebot@v2 (or v1.1, or whichever!).

**Comment by tbeswick96** [comment]
Any eta for this?

**Comment by JamesSingleton** [comment]
@goanpeca,

Any update on when this would potentially get merged and released?

**Comment by goanpeca** [comment]
Not my call 🤷‍♂

**Comment by JamesSingleton** [comment]
@chrispat maybe you can shed some light on this?

**Comment by ces131** [comment]
Really want to add this to a few repos but can't until multiple exempt labels are added - any updates??

**Comment by LekoArts** [comment]
As mentioned by me in another PR already: Feel free to use our (Gatsby) fork:
https://github.com/gatsbyjs/stale

**Comment by Gustry** [comment]
> As mentioned by me in another PR already: Feel free to use our (Gatsby) fork:
> https://github.com/gatsbyjs/stale

Thanks, I will give it a try!

It's somehow sad that GitHub (which owns this actions repository) doesn't look at PR on their own platforms.

**Comment by hross** [comment]
This repo is getting pretty stale and I'm taking it over to clean it up and get some of these suggestions merged. I like this one @ryanwilsonperkin if you want to add the comma parsing we can get it merged and pushed with the next release (soon).

And yep @Gustry it's sad this repo got neglected for so long 😢 (but we will rehab it shortly).

**Comment by ryanwilsonperkin** [comment]
Hey @hross thanks for taking over! I'll push an update to switch this over to comma-parsing instead.

**Comment by qJake** [comment]
@ryanwilsonperkin Thanks for the quick turnaround!

@hross Thanks for taking this over - really hoping to see a release very soon with this functionality in it! ❤️

**Comment by hross** [comment]
this looks good if you want to resolve the conflicts I'll merge it.

**Comment by ryanwilsonperkin** [comment]
On it

**Comment by ryanwilsonperkin** [comment]
@hross ready 🎉

**Comment by hross** [comment]
@ryanwilsonperkin yes you need to run `npm run all` (runs linting) or `npm run pack` to generate the new dist file.

**Comment by hross** [comment]
(I'll update the readme with better instructions)

**Comment by JamesSingleton** [comment]
@hross, can't wait for this to go in!

**Pull request merged**